### PR TITLE
ui: add ndb search to universe view

### PR DESCRIPTION
**Pull request by ericholguin**
This PR adds the NDB search functionality from the pull down search in the posting timeline to the universe view.

https://v.nostr.build/LtG2oHq7kUUPuMKX.mp4

Changelog-Added: Added NDB search functionality to the universe view

### CODE REVIEW

**Comment by jb55** [comment]
I didn't see this until now... will test soon after these ios18 fixes land

**Comment by jb55** [comment]
testing this now

File: damus/Views/SearchResultsView.swift [hunk 1]
```diff
@@ -51,7 +53,35 @@ struct InnerSearchResults: View {
...
+                Text("Notes", comment: "Navigation link to search text.")
+            }
+            .padding(.horizontal, 15)
+            .padding(.vertical, 5)
+            .background(DamusColors.neutral1)
+            .cornerRadius(20)
+            .overlay(
+                RoundedRectangle(cornerRadius: 20)
+                    .stroke(DamusColors.neutral3, lineWidth: 1)
+            )
```
Comment (danieldaquino) [inline damus/Views/SearchResultsView.swift:84]: We should define this gray button style as a reusable view, to reduce code duplication between `TextSearch` and `HashtagSearch`

File: damus/Views/SearchResultsView.swift [hunk 2]
```diff
@@ -88,8 +118,11 @@ struct InnerSearchResults: View {
             case .naddr(let naddr):
                 SearchingEventView(state: damus_state, search_type: .naddr(naddr))
             case .multi(let multi):
-                VStack {
-                    HashtagSearch(multi.hashtag)
+                VStack(alignment: .leading) {
+                    HStack {
+                        HashtagSearch(multi.hashtag)
+                        TextSearch(multi.text)
+                    }
```
Comment (danieldaquino) [inline damus/Views/SearchResultsView.swift:125]: I feel like we should add more spacing between these buttons and the profile search results below it.

Right now the spacing between the button and the top looks much larger than the button and the profile results below it, making it look a bit asymmetrical.
Screenshot 2024-09-27 at 11 27 11

**Review by danieldaquino — COMMENTED** [review]
Thank you @ericholguin! I really liked this new feature, I think it significantly improves search UX.

I just added two minor comments, but otherwise looks good to me!

**Comment by jb55** [comment]
merged and expanded the search results to 128!

**Pull request merged**